core/src/test/java/com/linecorp/armeria/server/streaming/JsonLinesTest.java

```diff
@@ -0,0 +1,160 @@
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.service("/seq/publisher",
+                    (ctx, req) -> JsonLines.fromPublisher(Flux.just("foo", "bar", "baz", "qux")))
+                    .service("/seq/stream",
+                            (ctx, req) -> JsonLines.fromStream(
+                                    Stream.of("foo", "bar", "baz", "qux"), MoreExecutors.directExecutor()))
+                    .service("/seq/custom-mapper",
+                            (ctx, req) -> JsonLines.fromPublisher(
+                                    Flux.just(new Pojo("Jon D", 21, Arrays.asList("Bmw", "Audi")),
+                                              new Pojo("Sarah D", 22, Arrays.asList("Tesla", "Honda"))),
+                                    new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT)))
+                    .service("/seq/single",
+                            (ctx, req) -> JsonLines.fromObject("foo"));
+            sb.disableServerHeader();
+            sb.disableDateHeader();
+        }
+    };
+
+    @Test
+    public void fromPublisherOrStreamMultiLineJson() {
+        final String record1 = "{\"name\":\"Jon D\",\"age\":21,\"cars\":[\"Bmw\",\"Audi\"]}\n";
+        final String record2 = "{\"name\":\"Sarah D\",\"age\":22,\"cars\":[\"Tesla\",\"Honda\"]}\n";
+        final WebClient client = WebClient.of(server.httpUri() + "/seq");
+        for (final String path : ImmutableList.of("/custom-mapper")) {
+            final HttpResponse response = client.get(path);
+            StepVerifier.create(response)
+                    .expectNext(ResponseHeaders.of(HttpStatus.OK,
+                            HttpHeaderNames.CONTENT_TYPE, MediaType.JSON_LINES))
+                    .assertNext(o ->
+                            ensureExpectedHttpData(o, record1, true))
+                    .assertNext(o ->
+                            ensureExpectedHttpData(o, record2, true))
+                    .assertNext(JsonLinesTest::assertThatLastContent)
+                    .expectComplete()
+                    .verify();
+        }
+    }
+
+    @Test
+    public void fromPublisherOrStream() {
+        final WebClient client = WebClient.of(server.httpUri() + "/seq");
+        for (final String path : ImmutableList.of("/publisher", "/stream")) {
+            final HttpResponse response = client.get(path);
+            StepVerifier.create(response)
+                    .expectNext(ResponseHeaders.of(HttpStatus.OK,
+                            HttpHeaderNames.CONTENT_TYPE, MediaType.JSON_LINES))
+                    .assertNext(o -> ensureExpectedHttpData(o, "foo", false))
+                    .assertNext(o -> ensureExpectedHttpData(o, "bar", false))
+                    .assertNext(o -> ensureExpectedHttpData(o, "baz", false))
+                    .assertNext(o -> ensureExpectedHttpData(o, "qux", false))
```

Comment:
Ditto, aggregate the result or use `HttpDecoder`.

Reply:
Done.